src/d3a/models/market/two_sided.py

```diff
@@ -264,9 +280,40 @@ def match_recommendation(self, recommended_list):
                     recommended_list, index+1, trade, bid_trade
                 )
 
+    @staticmethod
+    def validate_requirements_satisfied(
+            offer: Offer, bid: Bid, clearing_rate: float = None,
+            selected_energy: float = None) -> None:
+        """Validate if both trade parties satisfy each other's requirements.
+
+        :raises:
+            InvalidBidOfferPairException: Bid offer pair failed the validation
+        """
+        if ((offer.requirements or bid.requirements) and
+                not RequirementsSatisfiedChecker.is_satisfied(
+                    offer, bid, clearing_rate, selected_energy)):
+            # If no requirement dict is satisfied
+            raise InvalidBidOfferPairException(
+                f"OFFER: {offer} & BID: {bid} requirements failed the validation.")
+
+    @classmethod
+    def validate_authentic_bid_offer_pair(
```

Comment:
Nit: what do we mean by "authentic" here? For some reason every time I find this method I have a brain freeze, so maybe there's a more descriptive term we could use?

Reply:
BTW, I would add the description of the arguments in the docstring of this method. I would specifically describe what "selected_energy" and "clearing_rate" are (I think we have discussed this in a previous PR).

Reply:
I'm not sure, this was here before I touch the code, I also think that it can be renamed to validate_valid_bid_offer_pair for example, what do you think ?

Reply:
Or `validate_bid_offer_match`? Not sure though, you definitely have a clearer view of the context.

Reply:
Ah my suggested name has valid twice, you are right